internal/commands/publish.go

```diff
@@ -374,35 +384,44 @@ func endOfSupportFor(publishDate time.Time) string {
 	return endOfNinthMonth.Format(publishDateFormat)
 }
 
-func (p Publish) attachLicenseFile(slug string, releaseID int, version *releaseVersion) (string, error) {
+func (p Publish) attachLicenseFiles(slug string, releaseID int, version *releaseVersion) ([]string, error) {
+	var attachedLicenseFileNames []string
+
 	if version.IsGA() {
 		productFiles, err := p.PivnetProductFilesService.List(slug)
 		if err != nil {
-			return "", err
+			return attachedLicenseFileNames, err
 		}
 
-		productFile, found := findMatchingOSL(productFiles, version)
+		licenseFiles, found := findMatchingLicenseFiles(productFiles, version)
 		if !found {
-			return "", errors.New("required license file doesn't exist on Pivnet")
+			return attachedLicenseFileNames, errors.New("required license file doesn't exist on Pivnet")
 		}
 
-		err = p.PivnetProductFilesService.AddToRelease(slug, releaseID, productFile.ID)
-		if err == nil {
-			return productFile.Name, nil
-		} else {
-			return "", err
+		for _, licenseFile := range licenseFiles {
+			err = p.PivnetProductFilesService.AddToRelease(slug, releaseID, licenseFile.ID)
+
+			if err == nil {
+				attachedLicenseFileNames = append(attachedLicenseFileNames, licenseFile.Name)
+			} else {
+				return attachedLicenseFileNames, err
+			}
 		}
 	}
-	return "", nil
+	return attachedLicenseFileNames, nil
 }
 
-func findMatchingOSL(productFiles []pivnet.ProductFile, version *releaseVersion) (pivnet.ProductFile, bool) {
+func findMatchingLicenseFiles(productFiles []pivnet.ProductFile, version *releaseVersion) ([]pivnet.ProductFile, bool) {
+	var matchingLicenseFiles []pivnet.ProductFile
+	foundLicenses := false
```

Comment:
This variable is not needed it is effectively the same as checking `len(matchingLicenseFiles)`. I don't think second return values gives us anything.

Reply:
Instead of matching license files. It think it might be clearer and less error prone if we list the product file IDs in the Kilnfile.

Reply:
Did you one better - we don't even need a bool return, so i just made the caller of findMatchingLicenseFiles do the length checking instead!